CspSubmissionPlugin.inc.php

```diff
@@ -2080,6 +2096,39 @@ public function submissionfilesuploadformValidate($hookName, $args) {
 					QUERY
 				);
 			break;
+			case '57': // PDF para publicação PT
+			case '58': // PDF para publicação EN
+			case '59': // PDF para publicação ES
+				// Quando é feito upload PDF para publicação, editores de XML recebem email de convite para produzir XML
+				$userGroupEditorXML = 25;
+				$request = \Application::get()->getRequest();
+				$submissionId = $request->getUserVar('submissionId');
+				$userDao = DAORegistry::getDAO('UserDAO');
+				$site = $request->getSite();
+				$context = $request->getContext();
+				$result = $userDao->retrieve(
+					<<<QUERY
+					SELECT u.email, u.user_id
+					FROM ojs.users u
+					LEFT JOIN user_user_groups g
+					ON u.user_id = g.user_id
+					WHERE  g.user_group_id = $userGroupEditorXML
+					QUERY
+				);
+
+				import('lib.pkp.classes.mail.MailTemplate');
+				while (!$result->EOF) {
+					$mail = new MailTemplate('PRODUCAO_XML');
+					$mail->addRecipient($result->GetRowAssoc(0)['email']);
+					$mail->params["acceptLink"] = $request->_router->_indexUrl."/".$request->_router->_contextPaths[0]."/$$\$call$$$/grid/users/stage-participant/stage-participant-grid/save-participant/submission?submissionId=$submissionId&userGroupId=$userGroupEditorXML&userIdSelected=".$result->GetRowAssoc(0)['user_id']."&stageId=5&accept=1";
```

Comment:
oh boy, that's a long line

Reply:
```suggestion
					$mail->params["acceptLink"] = $request->_router->_indexUrl .  "/" .
					$request->_router->_contextPaths[0] .
					"/$$\$call$$$/grid/users/stage-participant/stage-participant-grid/save-participant/submission?submissionId=$submissionId&userGroupId=$userGroupEditorXML&userIdSelected=" . 
					$result->GetRowAssoc(0)['user_id'] . 
					"&stageId=5&accept=1";
```

Reply:
uma outra coisa, essa string ta com uma variaveis ali no meio tbm? ficou meio confuso

Reply:
$submissionId? Sim. Não encontrei outro jeito do cara clicar num link para ser designado para a submissão. Aceito sugestões

Reply:
o que eu quis dizer foi mais de dar tudo num bolo só, dai fica confuso de ler, se eu fosse fazer, faria algo nessa linha: 
```
$mail->params["acceptLink"] = strtr(
    ":indexUrl/:contextPath/:call/grid/users/stage-participant/stage-participant-grid/save-participant/submission?" .
    "submissionId=:submissionId&userGroupId=:userGroupEditorXML&userIdSelected=:userId&stageId=:stageId&accept=:accept",
    [
        ":indexUrl" => $request->_router->_indexUrl,
        ":contextPath" => $request->_router->_contextPaths[0],
        ":call" => "$$\$call$$$",
        ":submissionId" => $submissionId,
        ":userGroupEditorXML" => $userGroupEditorXML,
        ":userId" => $result->GetRowAssoc(0)['user_id'],
        ":stageId" => 5,
        ":accept" => 1,
    ]
);
```

dai tudo que tem na string é só string, fixo, coisa de url, e parametros fica no array embaixo. As queries do doctrine fazem algo semelhante com query de banco.

Tem só esse call ali que eu nao sei se é aquilo mesmo, pelo tanto de "$", talvez de pra simplificar

Reply:
Beleza, vou aplicar! Obrigada!